src/error.c

```diff
@@ -33,6 +34,28 @@ b32 HasErrors() {
     return errorCollector.errorCount > 0;
 }
 
+#define tempErrorBufferLen KB(4)
+char *temporaryErrorBuffer;
+
+void InitErrorBuffers() {
+    temporaryErrorBuffer = malloc(tempErrorBufferLen);
+};
+
+// NOTE: this function is not threadsafe
+char *errorBuffPrintf(const char *fmt, ...) {
```

Comment:
`array.c` has a function `ArrayPrintf` that will use a stretchy buffer and grow as necissary, maybe better to use that?